src/Microsoft.AspNet.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -0,0 +1,206 @@
+        public UrlResolutionTagHelper(IUrlHelper urlHelper, IHtmlEncoder htmlEncoder)
+        {
+            UrlHelper = urlHelper;
+            HtmlEncoder = htmlEncoder;
+        }
+
+        /// <inheritdoc />
+        public override int Order
+        {
+            get
+            {
+                return int.MinValue;
+            }
+        }
+
+        protected IUrlHelper UrlHelper { get; }
+
+        protected IHtmlEncoder HtmlEncoder { get; }
+
+        /// <inheritdoc />
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            IEnumerable<string> attributeNames;
+            if (ElementAttributeLookups.TryGetValue(output.TagName, out attributeNames))
+            {
+                foreach (var attributeName in attributeNames)
+                {
+                    ProcessUrlAttribute(attributeName, output);
+                }
+            }
+
+            // itemid can be present on any HTML element.
+            ProcessUrlAttribute("itemid", output);
+        }
+
+        private void ProcessUrlAttribute(string attributeName, TagHelperOutput output)
+        {
+            IEnumerable<TagHelperAttribute> attributes;
+            if (output.Attributes.TryGetAttributes(attributeName, out attributes))
+            {
+                foreach (var attribute in attributes)
+                {
+                    string resolvedUrl;
+                    if (attribute.Value is string)
+                    {
+                        if (TryResolveUrl(
+                                (string)attribute.Value,
+                                tryEncodeApplicationPath: false,
+                                resolvedUrl: out resolvedUrl))
+                        {
+                            attribute.Value = resolvedUrl;
+                        }
+                    }
+                    else if (attribute.Value is HtmlString)
+                    {
+                        var htmlStringValue = ((HtmlString)attribute.Value).ToString();
+                        if (TryResolveUrl(
+                                htmlStringValue,
+                                tryEncodeApplicationPath: true,
+                                resolvedUrl: out resolvedUrl))
+                        {
+                            attribute.Value = new HtmlString(resolvedUrl);
+                        }
+                    }
+                }
+            }
+        }
+
+        private bool TryResolveUrl(string url, bool tryEncodeApplicationPath, out string resolvedUrl)
+        {
+            resolvedUrl = null;
+
+            if (url == null)
+            {
+                return false;
+            }
+
+            var urlLength = url.Length;
+
+            // Find the start of the potential application relative URL value.
+            var prefixEndIndex = -1;
+            while (++prefixEndIndex < urlLength && char.IsWhiteSpace(url[prefixEndIndex])) ;
```

Comment:
`char.IsWhiteSpace()` covers far more characters than the expected Latin1 set. Since newlines, tabs, and so on aren't supported in a URL, suggest just `' ' == url[prefixEndIndex]`.